tests/php/Publisher/FilesystemPublisherTest.php

```diff
@@ -246,6 +246,17 @@ public function testPurgeURLAfterSwitchingExtensions(): void
         $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'purge-me.php');
     }
 
+    public function testNoPublishOnDisallowedResponseCode(): void
+    {
+        $this->logOut();
+
+        FilesystemPublisher::config()->set('disallowed_status_codes', [404]);
+
+        $this->fsp->publishURL('not_really_there', true);
+        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'not_really_there.html');
+        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'not_really_there.php');
```

Comment:
```suggestion
        FilesystemPublisher::config()->set('disallowed_status_codes', [404]);
        $this->fsp->publishURL('not_really_there', true);
        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'not_really_there.html');
        $this->assertFileDoesNotExist($this->fsp->getDestPath() . 'not_really_there.php');
        
        FilesystemPublisher::config()->set('disallowed_status_codes', []);
        $this->fsp->publishURL('not_really_there', true);
        $this->assertFileDoesExist($this->fsp->getDestPath() . 'not_really_there.html');
        $this->assertFileDoesExist($this->fsp->getDestPath() . 'not_really_there.php');
```
Or something similar

Reply:
There's already a separate test above this which is testing that the page _is_ cached - no need to add it again.